File: howl-application/src/main/java/io/quartic/howl/DiskStorageBackendConfig.java
package io.quartic.howl;

import io.quartic.howl.storage.DiskStorageBackend;
import io.quartic.howl.storage.StorageBackend;

import java.nio.file.Paths;

public class DiskStorageBackendConfig implements StorageBackendConfig {
    private String dataDir;

    public void setDataDir(String dataDir) {
        this.dataDir = dataDir;
    }

    @Override
    public StorageBackend build() {
        return new DiskStorageBackend(Paths.get(dataDir));
    }
}

File: howl-application/src/main/kotlin/io/quartic/howl/DiskStorageBackendConfig.kt
package io.quartic.howl

import io.quartic.howl.storage.DiskStorageBackend
import java.nio.file.Paths

data class DiskStorageBackendConfig(val dataDir: String = "./data") : StorageBackendConfig {
    override fun build() = DiskStorageBackend(Paths.get(dataDir))
}
